library/ssl_tls13_client.c

```diff
@@ -1752,6 +1754,19 @@ static int ssl_tls13_write_change_cipher_spec( mbedtls_ssl_context *ssl )
     if( ret != 0 )
         return( ret );
 
+    switch( ssl->state )
```

Comment:
This change introduces a bug similar as the one described in https://github.com/ARMmbed/mbedtls-restricted/issues/882. Thus please revert this change. We will think through this when we will work on https://github.com/ARMmbed/mbedtls-restricted/issues/882.